ipi/utils/io/backends/io_xyz.py

```diff
@@ -69,65 +68,70 @@ def print_xyz(atoms, cell, filedesc=sys.stdout, title=""):
+    """Reads an XYZ-style file with i-PI style comments and returns data in raw format for further units transformation
+    and other post processing.
 
     Args:
         filedesc: An open readable file object from a xyz formatted file with i-PI header comments.
 
     Returns:
-        An Atoms object with the appropriate atom labels, masses and positions.
-        A Cell object.
+        i-Pi comment line, cell array, data (positions, forces, etc.), atoms names and masses
     """
 
-    natoms = filedesc.readline()
+    try:
+        natoms = list(islice(filedesc,1))[0]
+    except:
+        natoms = ""
     if natoms == "":
         raise EOFError("The file descriptor hit EOF.")
     natoms = int(natoms)
-    comment = filedesc.readline()
-    reabc = re.compile('# CELL.abcABC.: ([-0-9.Ee ]*) ').search(comment)
-    regenh = re.compile('# CELL.GENH.: ([-0-9.Ee ]*)').search(comment)
-    reh = re.compile('# CELL.H.: ([-0-9.Ee ]*)').search(comment)
+
+    comment = list(islice(filedesc,1))[0]
+
+    # Extracting cell
+    cell = [key.search(comment) for key in cell_re]
     usegenh = False
-    if reabc is not None:
-        a, b, c, alpha, beta, gamma = reabc.group(1).split()
+    if cell[0] is not None:    # abcABC
+        a, b, c, alpha, beta, gamma = cell[0].group(1).split()
         a = float(a)
         b = float(b)
         c = float(c)
         alpha = float(alpha) * np.pi/180
         beta = float(beta) * np.pi/180
         gamma = float(gamma) * np.pi/180
         h = mt.abc2h(a, b, c, alpha, beta, gamma)
-    elif reh is not None:
-        h = np.array(reh.group(1).split(), float)
+    elif cell[1] is not None:  # GENH
+        h = np.array(cell[1].group(1).split(), float)
         h.resize((3,3))
-    elif regenh is not None:
-        genh = np.array(regenh.group(1).split(), float)
+    elif cell[2] is not None:  # H
+        genh = np.array(cell[2].group(1).split(), float)
         genh.resize((3,3))
         invgenh = np.linalg.inv(genh)
         # convert back & forth from abcABC representation to get an upper triangular h
         h = mt.abc2h(*mt.genh2abc(genh))
         usegenh = True
-    else:
-        # defaults to unit box
+    else:                     # defaults to unit box
         h = mt.abc2h(1.0, 1.0, 1.0, np.pi/2, np.pi/2, np.pi/2)
-    cell = Cell(h)
-
-    qatoms = []
-    names = []
-    masses = []
-    iat = 0
-    while (iat < natoms):
-        body = filedesc.readline()
-        if body.strip() == "":
-            break
-        body = body.split()
-        name = body[0]
-        names.append(name)
-        masses.append(Elements.mass(name))
-        x = float(body[1])
-        y = float(body[2])
-        z = float(body[3])
+    cell = h
+
+    qatoms = np.zeros(3*natoms)
+    names = np.zeros(natoms,dtype='|S4')
+    masses = np.zeros(natoms)
+
+    # Extracting a time-frame information
+    data =list(islice(filedesc,natoms))
```

Comment:
Yeah, was referring to the fact that max_rows was introduced in numpy 1.10, and it's not widely available